app/models/reviewable.rb

```diff
@@ -534,22 +535,60 @@ def self.list_for(
     result
   end
 
+  def self.recent_list_with_pending_first(user)
+    order_clause = DB.sql_fragment(<<~SQL, pending: statuses[:pending])
+      CASE WHEN reviewables.status = :pending THEN 0 ELSE 1 END,
+      CASE WHEN reviewables.status = :pending THEN reviewables.score END DESC,
+      reviewables.created_at DESC
+    SQL
+    query = Reviewable
+      .includes(:created_by, :topic, :target)
+      .viewable_by(user, preload: false)
+      .except(:order)
+      .order(order_clause)
+      .limit(30)
+
+    min_score = Reviewable.min_score_for_priority
+    if min_score > 0
+      query = query.where(<<~SQL, min_score: min_score)
+        reviewables.score >= :min_score OR reviewables.force_review
+      SQL
+    end
+    query
+  end
+
   def serializer
     self.class.serializer_for(self)
   end
 
+  def basic_serializer
+    self.class.basic_serializer_for(self)
+  end
+
   def self.lookup_serializer_for(type)
     "#{type}Serializer".constantize
   rescue NameError
     ReviewableSerializer
   end
 
+  def self.lookup_basic_serializer_for(type)
+    "Basic#{type}Serializer".constantize
+  rescue NameError
+    BasicReviewableSerializer
+  end
```

Comment:
I know we use this existing pattern in the model but I find it very unnecessary. We can just have a hash with the reviewable type as the key and the serializer class as the value. There is no need to `constantize` here.